extensions/amp-video-docking/0.1/amp-video-docking.js

```diff
@@ -54,52 +68,80 @@ import {
 
 const TAG = 'amp-video-docking';
 
-/** @visibleForTesting @const {number} */
-export const MARGIN_MAX = 30;
+/** Ratio to viewport width to use when docked to corner. */
+export const CORNER_WIDTH_RATIO = 0.3;
 
-/** @visibleForTesting @const {number} */
-export const MARGIN_AREA_WIDTH_PERC = 0.04;
+/** Min width in pixels for corner area. */
+export const CORNER_WIDTH_MIN = 180;
 
-/** @visibleForTesting @const {number} */
-export const MIN_WIDTH = 180;
+/** Max amount of pixels to use for margins (it's relative to viewport.) */
+export const CORNER_MARGIN_MAX = 30;
 
-/** @private @const {number} */
-const MIN_VIEWPORT_WIDTH = 320;
+/** Max % of viewport width to use form margins. */
+export const CORNER_MARGIN_RATIO = 0.04;
 
-/** @private @const {number} */
-const FLOAT_TOLERANCE = 0.02;
+/** Min viewport width for dock to appear at all. */
+export const MIN_VIEWPORT_WIDTH = 320;
 
-/** @visibleForTesting @const {string} */
-export const BASE_CLASS_NAME = 'i-amphtml-video-docked';
+/** Min visible intersection ratio of inline box to undock when scrolling. */
+export const REVERT_TO_INLINE_RATIO = 0.85;
+
+/**
+ * Width/height of placeholder icon when the inline box is large.
+ * @visibleForTesting
+ */
+export const PLACEHOLDER_ICON_LARGE_WIDTH = 48;
 
-/** @visibleForTesting @const {number} */
-export const REVERT_TO_INLINE_RATIO = 0.7;
+/** Margin applied to placeholder icon when the inline box is large. */
+export const PLACEHOLDER_ICON_LARGE_MARGIN = 40;
 
-/** @enum */
-export const RelativeX = {LEFT: 0, RIGHT: 1};
+/** Width/height of placeholder icon when the inline box is small. */
+export const PLACEHOLDER_ICON_SMALL_WIDTH = 32;
 
-/** @enum */
-export const RelativeY = {TOP: 0, BOTTOM: 1};
+/** Margin applied to placeholder icon when the inline box is large. */
+export const PLACEHOLDER_ICON_SMALL_MARGIN = 20;
+
+/** @const {!Array<!./breakpoints.SyntheticBreakpointDef>} */
+export const PLACEHOLDER_ICON_BREAKPOINTS = [
+  {
+    className: 'amp-small',
+    minWidth: 0,
+  },
+  {
+    className: 'amp-large',
+    minWidth: 420,
+  },
+];
 
-/** @enum */
-export const Direction = {UP: 1, DOWN: -1};
+/** @visibleForTesting @const {string} */
+export const BASE_CLASS_NAME = 'i-amphtml-video-docked';
 
 /** @enum {string} */
 export const Actions = {DOCK: 'dock', UNDOCK: 'undock'};
 
-/** @visibleForTesting */
-export const DOCKED_TO_CORNER_SIZING_RATIO = 0.3;
+/** @enum {string} */
+export const DockTargetType = {
```

Comment:
Do we expect any more dock target types? If not, would a Boolean be sufficient here? `isSlotTaget`?

Reply:
Hmm, there's no plan to have more targets but I'd rather keep the disambiguation at the same abstract level via a type.